cookbook/case_studies/feature_engineering/feast_integration/feast_workflow.py

```diff
@@ -0,0 +1,235 @@
+import os
+from datetime import datetime, timedelta
+
+import joblib
+import pandas as pd
+from feast import (
+    Entity,
+    Feature,
+    FeatureStore,
+    FeatureView,
+    FileSource,
+    RepoConfig,
+    ValueType,
+)
+from feast.infra.offline_stores.file import FileOfflineStoreConfig
+from feast.infra.online_stores.sqlite import SqliteOnlineStoreConfig
+from flytekit import reference_task, task, workflow
+from flytekit.extras.sqlite3.task import SQLite3Config, SQLite3Task
+from flytekit.types.file import JoblibSerializedFile
+from flytekit.types.file.file import FlyteFile
+from flytekit.types.schema import FlyteSchema
+from sklearn.model_selection import train_test_split
+from sklearn.naive_bayes import GaussianNB
+from flytekit.configuration import aws
+
+
+# %%
+# We define the reference tasks. A :py:func:`flytekit.reference_task` references the Flyte tasks that have already been defined, serialized, and registered.
+# The primary advantage of using a reference task is to reduce the redundancy; we needn't define the task(s) again if we have multiple datasets that need to be feature-engineered.
+@reference_task(
+    project="flytesnacks",
+    domain="development",
+    name="feast_integration.feature_eng_tasks.mean_median_imputer",
+    version="{{ registration.version }}",
```

Comment:
@kumare3 The workflow is getting registered only when registering it twice. Does `registration.version` get populated only after the reference tasks get registered?

Reply:
I couldn't get this to work, so reverted this change for the time being.

Also, do we really need this for this demo?

Reply:
Yeah. I think so.

Reply:
Did it not work?

Reply:
No, it didn't. Now that the bulk of the work is done I can investigate this.

Samitha, can you expand on why you think having these tasks imported this way is important for this demo? I feel like it's already complicated enough and that we should focus on the interaction with the feast SDK.

Reply:
Rather than importing directly, I thought reference tasks would make more sense. Also, I couldn't serialize the tasks with the relative import (I may have to use `.` in front of the file name). We need not worry about the relative imports with reference tasks, so I included it.